colly.go

```diff
@@ -768,6 +779,19 @@ func (c *Collector) OnRequest(f RequestCallback) {
 	c.lock.Unlock()
 }
 
+// OnResponseHeaders registers a function. Function will be executed on every response
+// when headers and status are already received, but body is not yet read.
+//
+// Like in OnRequest, you can call Request.Abort to abort the transfer. This might be
+// useful if, for example, you're following all hyperlinks, but want to avoid
+// downloading files. Be aware that using this will prevent HTTP/1.1 connection reuse,
```

Comment:
Does it prevent connection reuse only in case of abort?

Reply:
Yes.

Reply:
Though I didn't test HTTP/2.0, _I think it should_ handle it gracefully.

Reply:
Great, thanks, could you update the PR to the latest master? And it's ready to merge

Reply:
I checked HTTP/2 closedly, just in case. We indeed don't lose the connection reuse in HTTP/2 case. When `Response.Body` is closed, a single stream is reset, and connection remains alive: https://github.com/golang/net/blob/6afb5195e5aab057fda82e27171243402346b0ad/http2/transport.go#L2077-L2104. I verified it in Wireshark as well.

Reply:
> Great, thanks, could you update the PR to the latest master? And it's ready to merge

Yes, in a couple of days.

Reply:
Awesome, thanks!